server/index.js

```diff
@@ -64,6 +64,188 @@ app.get('/api/products/:productId', (req, res, next) => {
+    .then(result => {
+      const cartItems = result.rows;
+      res.json(cartItems);
+    }).catch(err => {
+      console.error(err);
+      res.status(500).json({
+        error: 'An unexpected error occurred.'
+      });
+    });
+});
+
+app.get('/api/cart', (req, res, next) => {
+  console.log('inside api/cart');
+  console.log(req.session.cartId);
+
+  if (!req.session.cartId) {
+    res.json([]);
+  } else {
+    const sql =
+        `select "c"."cartItemId",
+        "c"."price",
+        "p"."productId",
+        "p"."image",
+        "p"."name",
+        "p"."shortDescription"
+        from "cartItems" as "c"
+        join "products" as "p" using ("productId")
+        where "c"."cartId" = $1`;
+    const value = [req.session.cartId];
+    db.query(sql, value)
+      .then(result => {
+        res.json(result.rows);
+      });
+  }
+});
+
+app.post('/api/cart', (req, res, next) => {
+  var proId = parseInt(req.body.productId);
+  var values = [proId];
+
+  // var price = req.body.price;
+  // const numberPrice = price * 100;
+  console.log('proId: ' + proId);
+  // console.log('numberPrice ' + price);
+
+  if (!Number.isInteger(proId) || proId < 0) {
+    res.status(400).json({
+      error: 'invalid id'
+    });
+    return;
+  }
+  const sql =
+  `select
+  "price"
+  from "products"
+  where "productId" = $1`;
+
+  db.query(sql, values)
+    .then(result => {
+      var rowPrice = result.rows;
+      var arrayPrice = rowPrice.map(a => a.price);
+      var indexPrice = arrayPrice[0];
+      // console.log(indexPrice);
+      if (result.rows.length === 0) {
+        throw new ClientError('oh no , an unexpected error occurred.', 404);
+      }
+
+      if (req.session.cartId) {
+        console.log('cart id exists');
+
+        const sql3 = `insert into "cartItems" ("cartId", "productId", "price")
+          values ($1, $2, $3)
+          returning "cartItemId"`;
+
+        const values3 = [req.session.cartId, proId, indexPrice];
+        return db.query(sql3, values3)
+          .then(result3 => {
+            // console.log('result3: ', result3);
+
+            const cartItemId = result3.rows[0].cartItemId;
+            const sql4 =
+              `select "c"."cartItemId",
+                  "c"."price",
+                  "p"."productId",
+                  "p"."image",
+                  "p"."name",
+                  "p"."shortDescription"
+              from "cartItems" as "c"
+              join "products" as "p" using ("productId")
+            where "c"."cartItemId" = $1`;
+            const values4 = [cartItemId];
+
+            return db.query(sql4, values4)
+              .then(result4 => {
+                // console.log(result4);
+                res.status(201).json(
+                  result4.rows[0]
+                );
+              });
+          });
+
+      } else {
+        const sqlCart =
+      ` insert into "carts" ("cartId", "createdAt")
+          values (default, default)
+          returning "cartId" `;
```

Comment:
If there is already a session.cartId, we don't want to make another cart, just add the item to the same cart from before.